lib/ansible/config/base.yml

```diff
@@ -1249,6 +1249,13 @@ GALAXY_SERVER:
   ini:
   - {key: server, section: galaxy}
   yaml: {key: galaxy.server}
+GALAXY_TOKEN:
+  default: null
+  description: "GitHub personnal access token"
```

Comment:
Should this be `Galaxy` instead of `GitHub`?

Reply:
Right now, the galaxy login command needs a github Token. Though I named the variable GALAXY_TOKEN thinking that it may support other identity providers in the future.

Reply:
@trishnaguha ah ok, I follow. :+1:

Reply:
The nomenclature is a bit confusing. Here's the workflow... For a user to log into Galaxy we require GitHub authentication. That means we need either a GitHub username/password or a GitHub personal access token. We hand the auth values off to the Galaxy API, which attempts the authentication with GitHub, and on success, returns a Galaxy token. The Galaxy token gets written to a file.

When we one day expand to include other SCMs besides GitHub, we'll presumably want to enable authentication via the new providers. We could extend the current workflow, and add support for more providers.

Or... We change this completely, and like other services, we create a way on the Galaxy web site for a user to generate a Galaxy authentication token. The user then stores that token in a file or environment variable. With this approach we eliminate the need for the user to pass GitHub auth values through the CLI to the Galaxy API. 

In the long run, the second approach sounds better, where we create a page on the Galaxy web site enabling users to generate a Galaxy API token. And so given that, using the term 'GALAXY_TOKEN' in this PR makes sense.

Reply:
AIUI the current implementation requires linking an ansible galaxy login to a github login as above, with a token or the user/pass credentials, which therefore grants ansible galaxy full access to anything in the repos accessible to that user, regardless of relevance, where the old method of pulling that information based on an internally-hosted requirements.yml file does not. Lets assume that ansible-galaxy always behaves and doesn't look into any repos it shouldn't do, for simplicity.

This is fine up to a point, for publicly sharing role code (which is the idea behind ansible-galaxy). But what about username and password, URL, and other secret information that *must* be stored privately (meaning access can't be entrusted to a third party company such as Ansible)? And yet you need to be able to reliably pull with ansible galaxy (with a credentialled pull, obviously!) in order for your code to actually function properly?

There is a workaround if you're not using ansible tower, rather you're using Jenkins pipelines, and you pre-pull your code using a credentialled git pull in the pipeline and then use "--extra-vars" to pull in the resulting files. However this completely lacks inherent dependency management (that requirements.yml methods did provide), which is a headache, because you have to very carefully sort your own dependencies out.

I dare say that any role that does real useful work (which makes it worth sharing) will at some point have to have some kind of confidential information attached to it. Whether you reach this point by starting a role from scratch with separate secret information or redact/stub out your existing confidential information before sharing the role code, the problem is the same. If you can't load your confidential values (group_vars etc) from ansible-galaxy independently of publicly shared role code, and thus can't easily both share the role code on it *and* use it to continue to use your own ansible roles, why would anyone bother sharing them in the first place? Sounds a bit like ansible/galaxy is shooting itself in the foot.

I can see this might be because of a marketing excercise to build up contacts/etc or some other business reason, but it doesn't work if it renders your product unusable for serious users.

So please lets have the ability to do credentialled pulls from github *without* having to tie up a github login to a galaxy one.